### PR TITLE
Add Brats back and convert to recipe+clientapi

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Fixes # .

### Description

Add Brats back under research folder, convert from learner and json config to recipe with clientapi. Apply changes from https://github.com/NVIDIA/NVFlare/pull/4026

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: research/brats18/result_stat/plot_tensorboard_events.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+def main():
+    plt.figure()
+    num_site = len(sites_fl)
+    i = 1
+    # add event files
+
+    data = {"Config": [], "Epoch": [], "Dice": []}
+
```
[the remark this answers is not in the document]
Comment (ZiyueXu77) [reply research/brats18/result_stat/plot_tensorboard_events.py]: removed

**Comment by ZiyueXu77** [comment]
build

**Comment by ZiyueXu77** [comment]
> ### Greptile Summary
> Reintroduces BraTS18 brain tumor segmentation example under the research folder, migrating from JSON-based Learner API to modern Recipe and Client API patterns. The implementation demonstrates federated learning with MONAI, including differential privacy support via SVTPrivacy filter.
> 
> Key changes:
> 
> * Implements client-side training with proper error handling for empty data loaders (`client.py:248-249`) and NaN validation metrics (`client.py:187-189`)
> * Uses Recipe API with `TransferType.DIFF` for efficient weight-only parameter transfer, required for DP filter compatibility
> * Provides centralized training mode via `--centralized` flag when `n_clients=1`, using combined `site-All.json` dataset
> * Includes comprehensive testing and plotting utilities for model evaluation
> * Properly handles division by zero in both training loop and test evaluation script
> 
> The migration maintains compatibility with the existing BraTS18 dataset structure while modernizing the codebase to use current NVFlare APIs.
> 
> ### Confidence Score: 4/5
> * This PR is safe to merge with minor considerations
> * The implementation properly handles edge cases (empty data loaders, NaN metrics) and follows NVFlare patterns. The code addresses previously reported division-by-zero issues. Minor concerns around copyright year (2026) and potential for clearer error messaging
> * No files require special attention - all error handling is properly implemented
> 
> ### Important Files Changed
> Filename	Overview
> research/brats18/client.py	Implements client-side training for BraTS18 segmentation with proper error handling for empty data loaders and validation metrics
> research/brats18/job.py	Configures federated learning job using Recipe API with proper SVTPrivacy integration and simulation environment setup
> research/brats18/model.py	Simple wrapper around MONAI SegResNet that stores constructor args for NVFlare serialization
> research/brats18/result_stat/brats_3d_test_only.py	Model testing script with proper division-by-zero guards for metric computation across validation samples
> ### Sequence Diagram
> ```mermaid
> sequenceDiagram
>     participant User
>     participant JobScript as job.py
>     participant SimEnv as Simulation Environment
>     participant Server as FL Server
>     participant Client as client.py
>     participant Model as BratsSegResNet
>     participant DataLoader as MONAI DataLoader
> 
>     User->>JobScript: Execute with args (n_clients, num_rounds, etc.)
>     JobScript->>Model: create_brats_model()
>     Model-->>JobScript: BratsSegResNet instance
>     JobScript->>JobScript: Configure FedAvgRecipe with DIFF transfer
>     alt enable_dp flag set
>         JobScript->>JobScript: Add SVTPrivacy filter
>     end
>     JobScript->>SimEnv: Create simulation environment
>     JobScript->>SimEnv: recipe.execute()
>     
>     loop For each FL round (num_rounds)
>         SimEnv->>Server: Coordinate round
>         Server->>Client: Send global model (params)
>         Client->>Client: flare.receive()
>         Client->>Model: Load model weights
>         Client->>DataLoader: Build train/valid dataloaders
>         DataLoader-->>Client: Train and validation batches
>         
>         Client->>Model: Validate with global model
>         Model-->>Client: Global validation metric
>         Client->>Client: Log val_metric_global_model
>         
>         loop For aggregation_epochs
>             Client->>Model: Train on local data
>             Model-->>Client: Training loss
>             Client->>Client: Check len(train_loader) > 0
>             alt train_loader is empty
>                 Client->>Client: Raise ValueError
>             end
>         end
>         
>         Client->>Client: flare.send() with weight diff
>         Client->>Server: Send trained weights + metrics
>         
>         alt enable_dp
>             Server->>Server: Apply SVTPrivacy filter to weights
>         end
>         
>         Server->>Server: Aggregate client updates (FedAvg)
>     end
>     
>     Server-->>JobScript: Training complete
>     JobScript-->>User: Job status and results path
> ```

regarding copyright year: all existing files, even modified, will keep their original copyright year, all new files will have the current year when they are created. Please keep this in mind for future reviews.
